src/dynimage_serde.rs

```diff
@@ -0,0 +1,182 @@
+                        }
+                    }
+                }
+
+                deserializer.deserialize_identifier(FieldVisitor)
+            }
+        }
+
+        struct SerialBufferVisitor;
+        struct DynamicSerialImage {
+            dtype: String,
+            data: String,
+        }
+
+        impl<'de> Visitor<'de> for SerialBufferVisitor {
+            type Value = DynamicSerialImage;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("struct DynamicSerialImage")
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::SeqAccess<'de>,
+            {
+                let dtype: String = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(0, &self))?;
+                let data: String = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(1, &self))?;
+                Ok(DynamicSerialImage { dtype, data })
+            }
+
+            fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::MapAccess<'de>,
+            {
+                let mut data: Option<String> = None;
+                let mut dtype: Option<String> = None;
+
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Dtype => {
+                            if dtype.is_some() {
+                                return Err(de::Error::duplicate_field("dtype"));
+                            }
+                            dtype = Some(map.next_value()?);
+                        }
+                        Field::Data => {
+                            if data.is_some() {
+                                return Err(de::Error::duplicate_field("data"));
+                            }
+                            data = Some(map.next_value()?);
+                        }
+                    }
+                }
+
+                let dtype = dtype.ok_or_else(|| de::Error::missing_field("dtype"))?;
+                let data = data.ok_or_else(|| de::Error::missing_field("data"))?;
+
+                Ok(DynamicSerialImage { dtype, data })
+            }
+        }
+
+        const FIELDS: &[&str] = &["dtype", "data"];
+        let res =
+            deserializer.deserialize_struct("DynamicSerialImage", FIELDS, SerialBufferVisitor)?;
+
+        let mut imgdata = STANDARD_NO_PAD.decode(res.data.as_bytes()).map_err(|e| {
+            de::Error::custom(format!("Failed to decode base64 string: {}", e))
+        })?;
+        let imgfmt = match res.dtype.as_str() {
+            "png" => Ok(ImageFormat::Png),
+            "tiff" => {
+                use flate2::read::ZlibDecoder;
+                use std::io::Read;
+                let mut decoder = ZlibDecoder::new(imgdata.as_slice());
```

Comment:
It would be nice to use compression in the TIFF case, but I think that should be added as a feature to the tiff crate rather than doing it specially here.